modin/experimental/core/execution/native/implementations/omnisci_on_native/test/utils.py

```diff
@@ -88,21 +90,52 @@ def align_datetime_dtypes(*dfs):
     Passed Modin frames may be casted to pandas in the result.
     """
     datetime_cols = {}
+    time_cols = set()
     for df in dfs:
         for col, dtype in df.dtypes.items():
             # If we already decided to cast this column to DateTime no more actions are needed
             if col not in datetime_cols and is_datetime64_any_dtype(dtype):
                 datetime_cols[col] = dtype
-
-    casted_dfs = (
+            # datetime.time is considered to be an 'object' dtype in pandas that's why
+            # we have to explicitly check the values type in the column
+            elif (
+                dtype == np.dtype("O")
+                and col not in time_cols
+                # OmniSci has difficulties with empty frames, so explicitly skip them
+                # https://github.com/modin-project/modin/issues/3428
+                and len(df) > 0
+                and all(
+                    isinstance(val, datetime.time) or pandas.isna(val)
+                    for val in df[col]
+                )
+            ):
+                time_cols.add(col)
+
+    if len(datetime_cols) == 0 and len(time_cols) == 0:
+        return dfs
+
+    def convert_to_time(value):
+        """Convert passed value to `datetime.time`."""
+        if isinstance(value, datetime.time):
+            return value
+        elif isinstance(value, str):
+            return datetime.time.fromisoformat(value)
+        else:
+            return datetime.time(value)
+
+    casted_dfs = []
+    for df in dfs:
         # OmniSci has difficulties with casting to certain dtypes (i.e. datetime64),
-        # so casting it to pandas before doing 'astype'
-        tuple(try_cast_to_pandas(df).astype(datetime_cols) for df in dfs)
-        # This is required so we don't try to cast empty OmniSci frames to pandas:
-        # https://github.com/modin-project/modin/issues/3428
-        if len(datetime_cols)
-        else dfs
-    )
+        # so casting it to pandas
+        pandas_df = try_cast_to_pandas(df)
+        if len(datetime_cols):
+            pandas_df = pandas_df.astype(datetime_cols)
+        if len(time_cols):
```

Comment:
```suggestion
        if datetime_cols:
            pandas_df = pandas_df.astype(datetime_cols)
        if time_cols:
```
reads better, works faster (though I don't think one can measure how much faster - the difference is too small, so main point - reads better)

Reply:
fixed